File: frontend/src/api/customModel.ts
/**
 * 前端模型配置API
 */
import api from './index'

export interface CustomModel {
  id: number
  name: string
  display_name: string
  base_url: string
  default_model: string
  description?: string
  enabled: boolean
  created_at: string
  updated_at?: string
}

export interface CustomModelCreate {
  name: string
  display_name: string
  base_url: string
  api_key: string
  default_model: string
  description?: string
}

export interface TestConnectionResult {
  success: boolean
  message: string
  response?: string
}

export const customModelApi = {
  /**
   * 获取自定义模型列表
   */
  list: () => {
    return api.get<CustomModel[]>("/models/custom/")
  },

  /**
   * 创建自定义模型
   */
  create: (data: CustomModelCreate) => {
    return api.post<CustomModel>("/models/custom/", data)
  },

  /**
   * 更新自定义模型
   */
  update: (id: number, data: Partial<CustomModelCreate>) => {
    return api.put<CustomModel>(`/models/custom/${id}`, data)
  },

  /**
   * 删除自定义模型
   */
  delete: (id: number) => {
    return api.delete(`/models/custom/${id}`)
  },

  /**
   * 启用/禁用模型
   */
  toggle: (id: number) => {
    return api.post(`/models/custom/${id}/toggle`)
  },

  /**
   * 测试模型连接
   */
  test: (id: number) => {
    return api.post<TestConnectionResult>(`/models/custom/${id}/test`)
  },

  /**
   * 测试连接（创建前）
   */
  testConnection: (data: CustomModelCreate) => {
    return api.post<TestConnectionResult>("/models/custom/test-connection", data)
  },

  /**
   * 获取所有可用模型（包括内置和自定义）
   */
  getAllAvailable: () => {
    return api.get("/models/custom/all/available")
  }
}

File: backend/app/models/custom_model.py
"""
自定义模型配置
"""
from tortoise import fields
from tortoise.models import Model


class CustomModel(Model):
    """用户自定义模型配置表"""
    
    id = fields.IntField(pk=True, description="主键ID")
    name = fields.CharField(max_length=50, unique=True, description="模型标识（如 kimi）")
    display_name = fields.CharField(max_length=100, description="显示名称（如 Kimi AI）")
    base_url = fields.CharField(max_length=500, description="API 地址")
    api_key = fields.CharField(max_length=500, description="API Key")
    default_model = fields.CharField(max_length=100, description="默认模型名")
    description = fields.CharField(max_length=500, null=True, description="模型描述")
    enabled = fields.BooleanField(default=True, description="是否启用")
    created_at = fields.DatetimeField(auto_now_add=True, description="创建时间")
    updated_at = fields.DatetimeField(auto_now=True, description="更新时间")
    
    class Meta:
        table = "custom_models"
        table_description = "用户自定义AI模型配置表"
    
    def __str__(self):
        return f"CustomModel({self.name}, {self.display_name})"
